### PR TITLE
Recognize STA\MTA Attribute For Main Function

### PR DESCRIPTION
**Pull request by Anipik**
Fixes - #13688 
Tests PR - dotnet/corefx#26001
The First Commit is the reverted change #15586 
The last commit contains the changes to fix the error which was breaking the tests in master branch.

The tests marked with STAThread Attribute were failing in the test runs when the main thread was not able to run due to blocking or any other reason. The program was returning the exitCode as HOST_E_EXITPROCESS_TIMEOUT rather than its actual exitcode . We changed the code to return the actual Exitcode.

### CODE REVIEW

File: src/vm/ceemain.cpp [hunk 1]
```diff
@@ -2074,15 +2074,8 @@ DWORD WINAPI EEShutDownProcForSTAThread(LPVOID lpParameter)
     {
         action = eRudeExitProcess;
     }
-    UINT exitCode;
-    if (g_fWeOwnProcess)
```
Comment (danmosemsft) [inline src/vm/ceemain.cpp]: g_fWeOwnProcess is dead right? it's always false. remove it throughout.

**Comment by danmosemsft** [comment]
We spent some time figuring out what the problem was. What it was is not obvious from your change. That means you are forcing your code reviewer and future developers to figure it out for themselves, which is unhelpful. You should include in your description/top comment a detailed and accurate explanation of the theory as much as we understand it including explaining why it was sporadic and why we chose this fix.

**Comment by danmosemsft** [comment]
You should also be explicit that the first commit is simply a `git revert` of the revert that @jkotas did, which means the code reviewer does not need to look carefuly at it.

**Comment by danmosemsft** [comment]
Also please include a link to the issue in all PR's

**Comment by jkotas** [comment]
> You should also be explicit that the first commit is simply a git revert

The best way to do that is to `git revert` the revert commit. It will make it obvious that it is the original change and reduce risk of human-made mistakes.

File: src/vm/ceemain.cpp (hunk 1 above)
Comment (jkotas) [reply src/vm/ceemain.cpp]: Also, we should do the EEShutDownProcForSTAThread dance only that caused the bug only if `CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_FinalizeOnShutdown)` is set (it is legacy config switch that we may remove at some point).  EEShutDownProcForSTAThread was a workaround for deadlocks caused by finalization on shutdown. We got rid of finalization on shutdown, and so we should not have these deadlocks anymore.

File: src/vm/ceemain.cpp [hunk 2]
```diff
@@ -2176,7 +2167,7 @@ void STDMETHODCALLTYPE EEShutDown(BOOL fIsDllUnloading)
     }
 
 #ifdef FEATURE_COMINTEROP
-    if (!fIsDllUnloading && IsThreadInSTA())
+    if (!fIsDllUnloading && IsThreadInSTA() && CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_FinalizeOnShutdown))
```
Comment (jkotas) [inline src/vm/ceemain.cpp]: The `EXTERNAL_FinalizeOnShutdown` check should be done first, before `IsThreadInSTA`.

**Pull request merged**